src/win/pipe.c

```diff
@@ -2602,6 +2602,9 @@ int uv_pipe_pending_count(uv_pipe_t* handle) {
 
 
 int uv_pipe_getsockname(const uv_pipe_t* handle, char* buffer, size_t* size) {
+  if (buffer == NULL || size == NULL || *size == 0)
```

Comment:
Note: here I added the checks in each function rather than in `uv__pipe_getname` because there is a code path that derefs `size`.